prboom2/src/st_stuff.c

```diff
@@ -421,6 +421,8 @@ void ST_SetScaledWidth(void)
       break;
   }
 
+  ST_SCALED_WIDTH = (ST_SCALED_WIDTH + 3) & (int)~3;
```

Comment:
If the bar isn't a multiple of 4 in width, then this rounds up. If it rounds up to the screenwidth, then the offset will be 0. Then the side-filling code thinks "there's nothing to fill" and does a no-op. But actually there are <4 pixels empty. Then if you're in opengl it applies palette alpha into the empty region, but doesn't refresh the area when drawing the bar because it "isn't there", so the color gets stuck until another palette effect.

Is it an engine limitation that it has to be 4?

Reply:
I guess, technically it needs to be an even number, because it is later subtracted from another even number (i. e. `SCREENWIDTH`) and the difference divided by two to achieve `ST_SCALED_OFFSETX`:
```
ST_SCALED_OFFSETX = (SCREENWIDTH - ST_SCALED_WIDTH) / 2;
```
Obviously, you'll want `(SCREENWIDTH - ST_SCALED_WIDTH)` to be even, but for some reason I wanted `ST_SCALED_OFFSETX` to be an even number as well. I cannot remember the exact reason, though.